test/e2e/framework/framework.go

```diff
@@ -223,3 +225,32 @@ func (f *Framework) DeleteGateway(cluster framework.ClusterIndex, name string) {
 		return nil, err
 	}, framework.NoopCheckResult)
 }
+
+func (f *Framework) GetGatewayInformer(cluster framework.ClusterIndex) (cache.SharedIndexInformer, chan struct{}) {
+	stopCh := make(chan struct{})
+	informerFactory := externalversions.NewSharedInformerFactory(SubmarinerClients[cluster], 0)
+	informer := informerFactory.Submariner().V1().Gateways().Informer()
+	go informer.Run(stopCh)
+	Expect(cache.WaitForCacheSync(stopCh, informer.HasSynced)).To(BeTrue())
+	return informer, stopCh
+}
+
+func GetDeletionChannel(informer cache.SharedIndexInformer) chan string {
+	dh := simpleDeletionHandler{
+		deletionChannel: make(chan string),
```

Comment:
Perhaps we should add a capacity to the channel, something high like 100, otherwise it's a one-shot.

Reply:
:+1: 

This use case is just 1 that we will be looking for, but just in case,